Add task branch specific changes

Asfode1 opened the pull request.

Asfode1 committed:
Add task branch specific changes

Asfode1 committed:
Fix go.mod: remove problematic dependency and update go.sum

Asfode1 committed:
Fix linter errors: add proper error handling in tests

dirtymew reviewed (changes requested):
Привет Артем

по работе есть ряд замечаний: 
- в PR нет изменений по заданию
- должен быть один файл с двумя job
- нужно показать что публикация на dockerhub работает по git tag

Asfode1 committed:
feat: add unified CI/CD pipeline with test and docker jobs

- Create single workflow file with two jobs: test and docker
- Test job: linting, testing, and building application
- Docker job: publish to DockerHub only on git tags (v*)
- Remove separate workflow files
- Add comprehensive documentation
- Configure Docker build for linux/amd64 and linux/arm64 platforms

Asfode1 committed:
docs: update documentation and finalize CI/CD pipeline

- Update TASK_CHANGES.md with complete implementation details
- Enhance DOCKER_DEPLOYMENT.md with testing instructions
- Add FINAL_CHECKLIST.md for pre-submission verification
- Optimize ci-cd.yml workflow configuration

Asfode1 committed:
docs: add tag deployment demonstration

dirtymew reviewed (changes requested):
Привет Артем

публикация на dockerhub завершилась с ошибками

см https://github.com/Asfode1/workflow/actions/runs/17749735339/job/50442228437

скорее всего что-то не так с доступом

Asfode1 committed:
docs(ci): update docs and workflow before tag trigger

Asfode1 committed:
ci: fix DockerHub login to use username asfodel and image name

Asfode1 committed:
ci: set DockerHub username/image to asfode1

Asfode1 committed:
fix: disable CGO in Docker build for modernc.org/sqlite

dirtymew reviewed (approved):
Артем 

все ок - работа принята
